ethexe/cli/src/tests.rs

```diff
@@ -0,0 +1,328 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Integration tests.
+
+use crate::service::Service;
+use alloy::node_bindings::Anvil;
+use anyhow::{anyhow, Result};
+use ethexe_common::{db::CodesStorage, events::BlockEvent};
+use ethexe_db::{Database, MemDb};
+use ethexe_ethereum::{Ethereum, RouterQuery};
+use ethexe_observer::{Event, MockBlobReader, Observer, Query};
+use ethexe_processor::Processor;
+use ethexe_sequencer::Sequencer;
+use ethexe_signer::Signer;
+use ethexe_validator::Validator;
+use futures::StreamExt;
+use gprimitives::{ActorId, CodeId, H256};
+use std::sync::Arc;
+
+struct Listener {
+    receiver: tokio::sync::mpsc::Receiver<Event>,
+    _handle: tokio::task::JoinHandle<()>,
+}
+
+impl Listener {
+    pub fn new(mut observer: Observer) -> Self {
+        let (sender, receiver) = tokio::sync::mpsc::channel::<Event>(8 * 1024 * 1024);
```

Comment:
looks like overkill

Reply:
I think it's not overkill because we have CodeLoaded event which contains code bytes. So, our limits for code size is 512 kBytes,  so 16 CodeLoaded events may overload this buffer.

Reply:
but this is an event channel, not bytes (`channel::<T = Event>`)

Reply:
You are right. Set it to 1024